src/routes/s3/s3.controller.ts

```diff
@@ -217,14 +223,76 @@ export class S3Controller {
   @ApiResponse({ status: 200, description: 'Policy applied successfully' })
   @ApiResponse({ status: 400, description: 'No policy provided' })
   @ApiResponse({ status: 500, description: 'Server error' })
-  async applyBucketPolicy(@Param('bucketName') bucketName: string, @Body() applyPolicyDto: ApplyPolicyDto): Promise<{ message: string } | { statusCode: number; error: string }> {
+  async applyBucketPolicy(@Body() applyPolicyDto: ApplyPolicyDto, @Param('bucketName') bucketName?: string): Promise<{ message: string } | { statusCode: number; error: string }> {
     if (!applyPolicyDto.policy) {
       return {
         statusCode: HttpStatus.BAD_REQUEST,
         error: 'No policy provided',
       };
     }
-    await this.s3Service.applyBucketPolicy(bucketName, applyPolicyDto.policy);
+    await this.s3Service.applyBucketPolicy(applyPolicyDto.policy, bucketName);
     return { message: 'Policy applied successfully' };
   }
+
+  @Get('signed-cookies/:key')
+  @ApiOperation({ summary: 'Generate CloudFront signed cookies for a resource' })
+  @ApiParam({ name: 'key', description: 'Object key (relative path in CDN)' })
+  @ApiResponse({ status: 200, description: 'Returns signed cookies' })
+  @ApiResponse({ status: 500, description: 'Server error' })
+  async getSignedCookies(@Param('key') key: string, @Res() res: Response): Promise<void> {
+    try {
+      const cdnUrl = process.env['CDN_URL']!;
+      const resourceUrl = `https://${cdnUrl}/${encodeURIComponent(key)}`;
+      const keyPairId = 'K3AUK8QB395VGT';
+      const privateKeyPath = path.resolve(__dirname, '../../../cloudfront-private-key.pem');
+
+      const expiresInSeconds = 600;
```

Comment:
That constant should be part of the class rather than in the method scope -- additionally, I would call this `sessionCookieTimeout` to better convey its purpose.